packages/mwp-core/src/util/launchDarkly.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+
+const HEADER_FASTLY_CLIENT_IP = 'fastly-client-ip';
+const HEADER_FASTLY_X_REGION = 'x-region';
+
+/**
+ * Populates a LaunchDarklyUser object with member and request properties.
+ *
+ * @param member  a member state object
+ * @param request a request object
+ */
+export const getLaunchDarklyUser = (
+	member: Object,
+	request: HapiRequest
+): LaunchDarklyUser => {
+	const { id, name, email, country } = member;
+
+	const ip = getRemoteIp(request);
+	const key = (id && id.toString()) || '0';
+	const anonymous = key === '0';
+	const custom = getCustomAttributes(request);
+
+	const user: LaunchDarklyUser = {
+		key,
+		anonymous,
+		custom,
+	};
+
+	if (ip) {
+		user.ip = ip;
+	}
+	if (name) {
+		user.name = name;
+	}
+	if (email) {
+		user.email = email;
+	}
+	if (country) {
+		user.country = country;
+	}
+
+	return user;
+};
+
+const getRemoteIp = (request: HapiRequest): ?string => {
```

Comment:
Since this isn't an LD-specific utility, could you put it in a new module, e.g. `util/requestinfo.js` or something similar? Definitely a useful function to have available

Reply:
@mmcgahan Good idea. I've moved parsing of remote IP and parsing geo location from `X-Region` header to a new module `requestUtils`